File: codeforces/AncientBerlandCircus.py
import sys 
import math
# x1 -> arr[0]; x2 -> arr[2]; x3 -> arr[4]
class Solution:
    def Thelenght(self, arr): 
        a = math.sqrt((arr[2] - arr[4])**2 + (arr[3] - arr[5])**2)
        b = math.sqrt((arr[0] - arr[4])**2 + (arr[1] - arr[5])**2)
        c = math.sqrt((arr[0] - arr[2])**2 + (arr[1] - arr[3])**2)

        return a, b, c
    def menstruation(self, a, b, c): 
        p = (a + b + c) / 2 
        S = math.sqrt(p*(p - a)*(p - b)*(p - c))
        R = (a * b * c)/(4 * S) 
        return R
    def gcd_float(self, a, b):
        if b < 1e-4: 
            return a
        return self.gcd_float(b, a - math.floor(a/b) * b)
    def Angles(self, a, b, c):
        cos_A = (b**2 + c**2 - a**2) / (2 * b * c)
        cos_B = (a**2 + c**2 - b**2) / (2 * a * c) 
        cos_C = (a**2 + b**2 - c**2) / (2 * a * b)

        A = math.acos(max(-1.0, min(1.0, cos_A)))
        B = math.acos(max(-1.0, min(1.0, cos_B)))
        C = math.acos(max(-1.0, min(1.0, cos_C)))

        a1 = 2 * A 
        b1 = 2 * B
        c1 = 2 * C 

        return self.gcd_float(a1, self.gcd_float(b1, c1))








def main():
    input_data = sys.stdin.read().split()

    if not input_data: 
        return 
    points = [float(x) for x in input_data]

    if len(points) != 6:
        return

    sol = Solution()
    a, b, c = sol.Thelenght(points)
    R = sol.menstruation(a, b, c)
    alpha = sol.Angles(a, b, c)


    n = (2 * math.pi) / alpha
    area = (n / 2) * (R ** 2) * math.sin(alpha)

    print(f"{area:.6f}")
    

if __name__  == "__main__": 
    main()

File: codeforces/AncientBerlandCircus.c
/**
 *      author: teejaze
 **/


#include <stdio.h>
#include <math.h>
#include <stdlib.h>


#define PI 3.14159265358979323846

typedef struct
{
    double a;
    double b; 
    double c;
} TriangleSides;

typedef struct
{
    double x; 
    double y;
} POINTS;

TriangleSides get_lengths(POINTS p[])
{
    TriangleSides side; 
    side.a = sqrt(pow(p[1].x - p[2].x, 2) + pow(p[1].y - p[2].y, 2));
    side.b = sqrt(pow(p[0].x - p[2].x, 2) + pow(p[0].y - p[2].y, 2)); 
    side.c = sqrt(pow(p[0].x - p[1].x, 2) + pow(p[0].y - p[1].y, 2));
    
    return side;
}

double gcd_float(double a, double b)
{
    if (fabs(b) < 1e-4)
        return a;
    return gcd_float(b, fmod(a, b));
}

double clamp(double v)
{
    if (v < -1.0)
        return -1.0;
    if (v > 1.0)
        return 1.0;
    return v; 
}

int main(int argc, char *argv[])
{
    if (argc > 1)
        if (freopen(argv[1], "r", stdin) == NULL)
        {
            printf("can't open file %s \n", argv[1]);
            return 1; 
        }

    if (argc > 2)
        freopen(argv[2], "w", stdout); 


   POINTS p[3]; 
   int i; 
   double semi, S, R;
   double angle_a, angle_b, angle_c;
   double alpha, area; 

   for (i = 0; i < 3; i++)
       if (scanf("%lf %lf", &p[i].x, &p[i].y) != 2)  
           break;

   TriangleSides sides = get_lengths(p); 

   semi = (sides.a + sides.b + sides.c) / 2.0;
   S = sqrt(semi * (semi - sides.a) * (semi - sides.b) * (semi - sides.c));
   R = (sides.a * sides.b * sides.c) / (4.0 * S); 
    

   angle_a = 2.0 * acos(clamp((sides.b * sides.b + sides.c * sides.c - sides.a * sides.a) / (2.0 * sides.b * sides.c)));
   angle_b = 2.0 * acos(clamp((sides.a * sides.a + sides.c * sides.c - sides.b * sides.b) / (2.0 * sides.a * sides.c))); 
   angle_c = 2.0 * acos(clamp((sides.a * sides.a + sides.b * sides.b - sides.c * sides.c) / (2.0 * sides.a * sides.b)));


   alpha = gcd_float(angle_a, gcd_float(angle_b, angle_c)); 
   area = ((2.0 * PI / alpha) * R * R * sin(alpha)) / 2.0; 

   printf("%.6f", area); 

    return 0;







}
